api/src/utils/csv-utils/csv-config-validation.interface.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * The CSV configuration interface
+ *
+ */
+export interface CSVConfig<THeader extends Uppercase<string> = Uppercase<string>> {
+  /**
+   * Record containing the static headers, their aliases, and the `validateCell` and `setCellValue` callbacks
+   * to be called for each static cell.
+   *
+   * @type {Record<THeader, { aliases: Uppercase<string> } & CSVHeaderConfig>}
+   */
+  staticHeadersConfig: Record<THeader, { aliases: Uppercase<string> } & CSVHeaderConfig>;
+  /**
+   * Contains the `validateCell` and `setCellValue` callbacks to be called for each dynamic cell.
+   *
+   * @type {CSVHeaderConfig | undefined}
```

Comment:
Might be worth explaining what we consider a static vs dynamic cell

Reply:
Oops I totally forgot to come back to this. Updated.